api/base/filters.py

```diff
@@ -433,14 +433,15 @@ def get_queryset_from_request(self):
     def param_queryset(self, query_params, default_queryset):
         """filters default queryset based on query parameters"""
         filters = self.parse_query_params(query_params)
-        queryset = set(default_queryset)
-
+        queryset = default_queryset
         if filters:
             for key, field_names in filters.iteritems():
                 for field_name, data in field_names.iteritems():
-                    queryset = queryset.intersection(set(self.get_filtered_queryset(field_name, data, default_queryset)))
-
-        return list(queryset)
+                    query_field_name = data['source_field_name']
+                    if query_field_name == 'tags':
```

Comment:
I added a hook for this a while back, `postprocess_query_param`, for this type of use case. See https://github.com/CenterForOpenScience/osf.io/blob/be40da61bb738b9c81d7b4d0f3cc2a81abff2463/api/nodes/views.py#L302-L311 for an example. Can you use that instead?